controllers/controllers/workloads/cfspace_controller.go

```diff
@@ -383,7 +386,7 @@ func (r *CFSpaceReconciler) finalizeCFApp(ctx context.Context, log logr.Logger,
 	}
 
 	done := make(chan bool, 1)
-	go waitForCFAppDeletion(ctx, log, namespace, done)
+	waitForCFAppDeletion(ctx, log, namespace, done)
```

Comment:
Sorry we missed this on the initial implementation but it looks like we should pull the timeout channel time.After out of the for loop or else it just resets the timer each iteration.

https://go.dev/play/p/DMnPRODlXmb is an example of the timeout not triggering.
https://go.dev/play/p/_fofq32LHLH is an example of the timeout working as expected.

Reply:
Good catch. Fixed